### PR TITLE
Feat/xr config

### PR DESCRIPTION
**Pull request by thestonefox**

### CODE REVIEW

File: Prefabs/[UnityXRCameraRig]/Scripts/UnityXRConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    using UnityEngine;
+    using UnityEngine.XR;
+
+    /// <summary>
+    /// Provides configuration for the Unity Engine in XR.
+    /// </summary>
+    public class UnityXRConfiguration : MonoBehaviour
+    {
+        /// <summary>
+        /// Represents the size of physical space available for XR.
```
[the remark this answers is not in the document]
Comment (thestonefox) [reply Prefabs/[UnityXRCameraRig]/Scripts/UnityXRConfiguration.cs]: This is the exact text unity use on the TrackingSpace enum, go figure

File: Prefabs/[UnityXRCameraRig]/Scripts/UnityXRConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        /// </summary>
+        [Tooltip("Represents the size of physical space available for XR.")]
+        public TrackingSpaceType trackingSpaceType = TrackingSpaceType.RoomScale;
+        /// <summary>
+        /// Automatically set the Unity Physics Fixed Timestep value based on the HMD render frequency.
+        /// </summary>
+        [Tooltip("Automatically set the Unity Physics Fixed Timestep value based on the HMD render frequency.")]
+        public bool lockPhysicsUpdateRateToRenderFrequency = true;
+
+        protected TrackingSpaceType currentType;
```
[the remark this answers is not in the document]
Comment (thestonefox) [reply Prefabs/[UnityXRCameraRig]/Scripts/UnityXRConfiguration.cs]: I think this should depend on a case by case basis.

If i change a bool in the inspector at runtime, I'd personally expect that to have an effect in the scene. Changing a tickbox and nothing happening would feel broken to me.

[the remark this answers is not in the document]
Comment (bddckr) [reply Prefabs/[UnityXRCameraRig]/Scripts/UnityXRConfiguration.cs]: Unity offers `OnValidate` for that. Additionally I would not expect this component to waste work with its Update despite only being used once to set these settings in most cases. 😄

[the remark this answers is not in the document]
Comment (thestonefox) [reply Prefabs/[UnityXRCameraRig]/Scripts/UnityXRConfiguration.cs]: Ok maybe it should use OnValidate then

**Pull request merged**